shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -297,12 +299,12 @@ public boolean handleKeyEvent(KeyEvent event) {
         final int selStart = Selection.getSelectionStart(mEditable);
         final int selEnd = Selection.getSelectionEnd(mEditable);
         final int character = event.getUnicodeChar();
-        if (selStart < 0 || selEnd < 0 || character == 0) {
+        if (character == 0) {
           return false;
         }
 
-        final int selMin = Math.min(selStart, selEnd);
-        final int selMax = Math.max(selStart, selEnd);
+        final int selMin = Math.max(Math.min(selStart, selEnd), 0);
```

Comment:
related to the question above, when is selStart or selEnd < 0 ?

Reply:
In this issue  case, the user use an TextInputConfiguration to connect the keyboard, but not use TextFiled, when the keyboard is number it will go to the `handleKeyEvent` method and cause is input a number then in to the `else branch` in this case the `selStart and selEnd will < 0` that cause the bug